sdl2/test/joystick_test.py

```diff
@@ -426,7 +425,8 @@ def test_SDL_JoystickNumButtons(joysticks):
             assert buttons == 4
 
 def test_SDL_JoystickUpdate(with_sdl):
-    # NOTE: Returns void, can't really test anything else
+    # TODO: This is not 100% safe, but in SDL2, JoystickUpdate returns void
+    # so we can't reliably detect error
     sdl2.SDL_JoystickUpdate()
     assert SDL_GetError() == b""
```

Comment:
I actually have an auto-use pytest fixture set up to call `SDL_ClearError` before and after each test is run to avoid error contamination across tests:

```python
@pytest.fixture(autouse=True)
def sdl_cleanup():
    sdl2.SDL_ClearError()
    yield
    sdl2.SDL_ClearError()
    gc.collect()
```

(the `gc.collect` is to immediately catch any double-free segfaults, which used to happen occasionally if memory ran low enough on certain tests before I rewrote the  `sdl2.ext` modules to be more robust)

Reply:
But you're the maintainer, and if you prefer to rely on the pytest fixture, do so.

Reply:
Actually, I do like the reasoning of making the call explicit for these marked-unsafe tests so it's perfectly clear at a glance where the error came from. I'll update the code accordingly.